src/components/Cards/CardsContainer.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+
+import ApiSubject from "../../../api/mockedApi";
+import { ICard } from "../../../api/mockedResponse";
+
+import Card from "./Card";
+import CardsCreationForm from "./CardsCreationForm";
+
+const CardsContainer: React.FC = () => {
+  const [cards, setCards] = React.useState<ICard[]>([]);
+
+  React.useEffect(() => {
+    ApiSubject.subscribe(setCards);
+    ApiSubject.dispatch();
+    return () => {
+      ApiSubject.unsubscribe(setCards);
+    }
+  }, []);
+
+  const onDelete = (id) => {
+    setCards(cards.filter(({ id: cardId }) => cardId !== id));
+  }
+
+  const onSubmit = ({ gender, title, price, imageUrl }) => {
+    const id = new Date().getTime();
+    setCards([...cards, { id, gender, title, price, imageUrl }]);
+  };
+
+  const renderCards = React.useMemo(() => {
+    if (!cards || !cards.length) {
+      return <div>No cards yet</div>;
+    }
+    return cards.map(({ gender, id, imageUrl, price, title }) => {
```

Comment:
What do you think about it?
```suggestion
    return cards.map((card: ICard) => {
```

Reply:
Fixed
https://github.com/BlackTruth/react/blob/80c18becffd0f0491d34262d50d4f802bad3da1e/src/components/Cards/CardsContainer.tsx#L41-L44